contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -215,25 +287,43 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
         require(recipient != address(0), "NativeTokenDestination: zero recipient address");
         require(amount != 0, "NativeTokenDestination: zero transfer value");
 
+        uint256 scaledAmount = _scaleTokens(amount, true);
+
```

Comment:
can you add comment if scaledAmount is zero we just return in the else block if there's `currentReserveImbalance`. Other option is to add an extra check if scaledAmount is zero before.

Reply:
refactored this a little bit. lmk if it makes sense.